src/expr/src/vector_op/cast.rs

```diff
@@ -183,7 +183,9 @@ pub fn i64_to_timestampz(t: i64) -> Result<i64> {
 pub fn timestampz_to_utc_string(elem: i64) -> String {
     // Just a meaningful representation as placeholder. The real implementation depends on TimeZone
     // from session. See #3552.
-    let instant = Utc.timestamp_nanos(elem * 1000);
+    let secs = elem / 1_000_000;
+    let nsecs = (elem % 1_000_000) * 1000;
+    let instant = Utc.timestamp_opt(secs, nsecs as u32).unwrap();
```

Comment:
* What if `elem` is negative? What would happen during the `usecs as u32` cast?
* Is the `unwrap()` guaranteed to never panic?

Reply:
1. secs will be negative.
2. no, it will panic if timestamp value is too big. This PR just makes it not as easy as before to panic

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e2f7bce89ffb0097e25f3039d8da1870